pkg/validation/components.go

```diff
@@ -103,7 +106,9 @@ func ValidateComponents(components []v1alpha2.Component) error {
 	for componentName, volumeMountNames := range processedVolumeMounts {
 		for _, volumeMountName := range volumeMountNames {
 			if !processedVolumes[volumeMountName] {
-				invalidVolumeMountsErr += fmt.Sprintf("\nvolume mount %s belonging to the container component %s", volumeMountName, componentName)
+				missingVolumeMountErr := fmt.Errorf("\nvolume mount %s belonging to the container component %s", volumeMountName, componentName)
```

Comment:
Outside the scope of this PR, but seeing `fmt.Errorf("\nvolume` makes me feel like `MissingVolumeMountError` should store `errs: []string` rather than `errMsg: string`.

Reply:
updated